alica_engine/src/engine/blackboard/ByteArray.cpp

```diff
@@ -0,0 +1,45 @@
+
+#include "engine/blackboard/ByteArray.h"
+
+#include <assert.h>
+#include <utility>
+#include <string.h>
+
+namespace alica {
+ByteArray::ByteArray(const int8_t* src, int32_t count)
+    : _size(count) {
+    assert(src != nullptr);
+    if (count > 0) {
+        _begin = static_cast<int8_t*>(malloc(count));
+        assert(_begin!=nullptr);
+        memcpy(_begin, src, count);
+    } else {
+        _begin = nullptr;
+    }
+}
+
+ByteArray::ByteArray(int32_t size)
+    : _size(size)
+{
+    assert(size > 0);
+    if(size> 0) {
+        _begin = static_cast<int8_t*>(malloc(size));
+    } else {
```

Comment:
I know this PR is merged already but I just observed that this else is redundant.